src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadExerciseResource.java

```diff
@@ -72,6 +74,12 @@ public FileUploadExerciseResource(FileUploadExerciseRepository fileUploadExercis
         if (fileUploadExercise.getId() != null) {
             throw new BadRequestAlertException("A new fileUploadExercise cannot already have an ID", ENTITY_NAME, "idexists");
         }
+
+        // Check if file pattern matches regex
+        Matcher filePatternMatcher = FILE_PATTERN.matcher(fileUploadExercise.getFilePattern());
```

Comment:
Since you use these three lines two times, can you please move it into a separate method?

Reply:
As far as I can see, you mean the empty line, the comment and the line after. I don't think it makes sense to create a separate method for those. 

If you mean the if-statement (or all 6 new lines), it might make sense to create a void method which does the validation and throws an exception and call it here because otherwise I will have to add  different additional lines and we again won't have consistency. (Check out the shortname validation in CourseResource).

Reply:
You are right it is not exactly three lines. But I meant the lines that you added
```
        Matcher filePatternMatcher = FILE_PATTERN.matcher(fileUploadExercise.getFilePattern());
       if (!filePatternMatcher.matches()) {
            return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName, "The specified file patterns are invalid", "filePatternInvalid")).body(null);
        }
```